### PR TITLE
[config] add gzip config to nginx

### PR DESCRIPTION
**Pull request by AZtheAsian**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -52,3 +52,9 @@ openwisp2_topology_save_snapshot_frequency:
     day: "*"
     hour: "23"
     minute: "30"
+openwisp2_nginx_ssl_config:
+  gzip: "on"
```
Comment (nemesisdesign) [inline defaults/main.yml]: use 4 space indentation as the rest of the file does

File: defaults/main.yml [hunk 2]
```diff
@@ -52,3 +52,9 @@ openwisp2_topology_save_snapshot_frequency:
...
+  gzip_comp_level: "6"
+  gzip_proxied: "any"
+  gzip_min_length: "1000"
+  gzip_types: "text/plain text/html image/svg+xml application/json application/javascript text/xml text/css application/xml application/x-font-ttf font/opentype"
```
Comment (nemesisdesign) [inline defaults/main.yml]: can you turn this long string into a list of items that is then reassembled in the nginx template?

Eg: in the template you can probably use `{{ ' '.join(gzip_types) }}`

File: templates/nginx.j2 [hunk 3]
```diff
@@ -27,6 +27,13 @@ server {
     add_header Strict-Transport-Security "max-age=31536000";
     add_header X-Content-Type-Options nosniff;
 
+    gzip {{ openwisp2_nginx_ssl_config.gzip }};
+    gzip_comp_level {{ openwisp2_nginx_ssl_config.gzip }};
+    gzip_proxied {{ openwisp2_nginx_ssl_config.gzip }};
+    gzip_min_length {{ openwisp2_nginx_ssl_config.gzip }};
+    gzip_proxied {{ openwisp2_nginx_ssl_config.gzip }};
+    gzip_types {{ openwisp2_nginx_ssl_config.gzip }};
```
Comment (nemesisdesign) [inline templates/nginx.j2]: all these lines refer to `gzip` while they should refer their respective correct key

**Comment by AZtheAsian** [comment]
@nemesisdesign got it fixed - sorry about that.

It works though!

got it

File: defaults/main.yml [hunk 4]
```diff
@@ -52,3 +52,9 @@ openwisp2_topology_save_snapshot_frequency:
     day: "*"
     hour: "23"
     minute: "30"
+openwisp2_nginx_ssl_config:
+    gzip: "on"
+    gzip_comp_level: "6"
+    gzip_proxied: "any"
+    gzip_min_length: "1000"
+    gzip_types: ["text/plain", "text/html", "image/svg+xml", "application/json", "application/javascript", "text/xml", "text/css", "application/xml", "application/x-font-ttf", "font/opentype"]
```
Comment (nemesisdesign) [inline defaults/main.yml]: use the extended format, eg:

```yaml
gzip_types:
    - item1
    - item2
```

**Comment by AZtheAsian** [comment]
@nemesisdesign like this?

File: templates/nginx.j2 [hunk 5]
```diff
@@ -27,6 +27,14 @@ server {
     add_header Strict-Transport-Security "max-age=31536000";
     add_header X-Content-Type-Options nosniff;
 
+    {% for gzip_variable in openwisp2_nginx_ssl_config %}
```
Comment (nemesisdesign) [inline templates/nginx.j2]: you could have named this `config` since this dict can be used also for non-gzip related variables now

**Pull request merged**